packages/@react-stately/select/src/useSelectState.ts

```diff
@@ -62,6 +62,7 @@ export function useSelectState<T extends object>(props: SelectStateOptions<T>):
   });
 
   let [isFocused, setFocused] = useState(false);
+  let isEmpty = useMemo(() => listState.collection.size === 0 || (listState.collection.size === 1 && listState.collection.getItem(listState.collection.getFirstKey()!)?.type === 'loader'), [listState.collection]);
```

Comment:
not great that stately has to know about this... perhaps we can refactor the isEmpty checks to centralize this logic